File: finanzas_home_web_final_consolidado/backend/app/schemas/tesoreria.py
from pydantic import BaseModel, Field

class CatalogoSimpleOut(BaseModel):
    id: int
    nombre: str
    codigo: str | None = None

class PagoDesdeCuentaOut(BaseModel):
    id: int
    fecha: str
    cuentaOrigen: str
    cuentaOrigenId: int | None = None
    pagadoA: str | None = None
    responsable: str | None = None
    responsableId: int | None = None
    rubroClasificacion: str | None = None
    rubroClasificacionId: int | None = None
    rubroAplicado: str | None = None
    rubroAplicadoId: int | None = None
    periodo: str | None = None
    periodoId: int | None = None
    descripcion: str | None = None
    monto: float
    referenciaExterna: str | None = None

class PagoDesdeCuentaUpsertIn(BaseModel):
    fecha: str
    cuentaOrigenId: int = Field(gt=0)
    pagadoA: str | None = None
    responsableId: int | None = None
    rubroClasificacionId: int | None = None
    rubroAplicadoId: int | None = None
    periodoId: int | None = None
    descripcion: str | None = None
    monto: float = Field(gt=0)
    referenciaExterna: str | None = None

File: finanzas_home_web_final_consolidado/frontend/src/types/tesoreria.ts
export type CatalogoSimple = {
  id: number;
  codigo?: string;
  nombre: string;
};

export type PagoDesdeCuenta = {
  id: number;
  fecha: string;
  cuentaOrigen: string;
  cuentaOrigenId?: number;
  pagadoA: string | null;
  responsable: string | null;
  responsableId?: number | null;
  rubroClasificacion: string | null;
  rubroClasificacionId?: number | null;
  rubroAplicado: string | null;
  rubroAplicadoId?: number | null;
  periodo: string | null;
  periodoId?: number | null;
  descripcion: string | null;
  monto: number;
  referenciaExterna: string | null;
};

export type PagoDesdeCuentaUpsert = {
  fecha: string;
  cuentaOrigenId: number;
  pagadoA: string | null;
  responsableId: number | null;
  rubroClasificacionId: number | null;
  rubroAplicadoId: number | null;
  periodoId: number | null;
  descripcion: string | null;
  monto: number;
  referenciaExterna: string | null;
};
